PR: https://github.com/odoo/odoo/pull/41206

From: 92472fbf9cabcc7c8a7bb5703bd5ac85ccd3a55e
From: nje-odoo
Date: 2019-12-31 12:17:26

Structural Changes: 10
Total Changes: 10

[IMP] lunch : make field translatable.

Now we can translate name and description field in lunch.product object
and we want that translation in kanban view of 'lunch.product.report'.

So we set related field in 'lunch.product.report' and remove those fields
from init().

Due to related field it will affect on performace of lunch.product.report
kanban view loading.

Task-2068989
closes odoo/odoo#41206

Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/lunch/models/lunch_alert.py
+++ b/addons/lunch/models/lunch_alert.py
@@ -19,8 +19,8 @@ class LunchAlert(models.Model):
     _description = 'Lunch Alert'
     _order = 'write_date desc, id'
 
-    name = fields.Char('Alert Name', required=True)
-    message = fields.Html('Message', required=True)
+    name = fields.Char('Alert Name', required=True, translate=True)
+    message = fields.Html('Message', required=True, translate=True)
 
     mode = fields.Selection([
         ('alert', 'Alert in app'),

--- a/addons/lunch/models/lunch_product.py
+++ b/addons/lunch/models/lunch_product.py
@@ -12,7 +12,7 @@ class LunchProductCategory(models.Model):
     _inherit = 'image.mixin'
     _description = 'Lunch Product Category'
 
-    name = fields.Char('Product Category', required=True)
+    name = fields.Char('Product Category', required=True, translate=True)
     company_id = fields.Many2one('res.company', default=lambda self: self.env.company)
     currency_id = fields.Many2one('res.currency', related='company_id.currency_id')
     topping_label_1 = fields.Char('Extra 1 Label', required=True, default='Extras')
@@ -89,9 +89,9 @@ class LunchProduct(models.Model):
     _inherit = 'image.mixin'
     _order = 'name'
 
-    name = fields.Char('Product Name', required=True)
+    name = fields.Char('Product Name', required=True, translate=True)
     category_id = fields.Many2one('lunch.product.category', 'Product Category', required=True)
-    description = fields.Text('Description')
+    description = fields.Text('Description', translate=True)
     price = fields.Float('Price', digits='Account', required=True)
     supplier_id = fields.Many2one('lunch.supplier', 'Vendor', required=True)
     active = fields.Boolean(default=True)
